### ISSUE TITLE
epoch or max_steps?

### ISSUE THREAD

**Issue by IcfBrest** [sjtuplayer/anomalydiffusion#91]
作者你好，我看项目里好像没有直接设置epoch而是只通过设置训练步数max_steps来控制训练次数，是这样吗？
不好意思，我是新手，如果问题有误的话请见谅。

**Comment by alwinsiemens** [comment]
Hi, I have a similar question: My average epoch-time is about 16 seconds on Nvidia A100 with the configuration file provided by you. Is that comparable to the avg epoch times from the paper-runs?

**Comment by alwinsiemens** [comment]
I have the similar question. Current experiments show me, that the iterations are dependend on the size of the dataset (you could try to play with max_steps with a smaller dataset). In the paper 300.000 iterations are mentioned for the first training. The Trainer uses per epoch 2331 images. 300.000 iterations would be on the MVTEC-AD dataset about 128 epochs I conclude.

**Comment by IcfBrest** [comment]
> I have the similar question. Current experiments show me, that the iterations are dependend on the size of the dataset (you could try to play with max_steps with a smaller dataset). In the paper 300.000 iterations are mentioned for the first training. The Trainer uses per epoch 2331 images. 300.000 iterations would be on the MVTEC-AD dataset about 128 epochs I conclude.

Thank you very much for your reply

**Comment by DqqHns** [comment]
> 我也有类似的问题。目前的实验表明，迭代次数取决于数据集的大小（你可以尝试使用较小的数据集来调整 max_steps）。论文中提到首次训练需要 30 万次迭代。训练器每个 epoch 使用 2331 张图像。我推断，在 MVTEC-AD 数据集上进行 30 万次迭代大约需要 128 个 epoch。

你好，可是我在单张3090上训练MVTEC-AD 数据集，第一步（异常图像生成）就已经花了5天了，目前还在训练，大概到320epoch了，这有点太久了吧，不清楚为什么？

**Comment by sjtuplayer** [comment]
> > 我也有类似的问题。目前的实验表明，迭代次数取决于数据集的大小（你可以尝试使用较小的数据集来调整 max_steps）。论文中提到首次训练需要 30 万次迭代。训练器每个 epoch 使用 2331 张图像。我推断，在 MVTEC-AD 数据集上进行 30 万次迭代大约需要 128 个 epoch。
> 
> 你好，可是我在单张3090上训练MVTEC-AD 数据集，第一步（异常图像生成）就已经花了5天了，目前还在训练，大概到320epoch了，这有点太久了吧，不清楚为什么？

Hi, you can stop at 30w iterations by setting the max_step, which may only take around 2-3 days on a single 3090 GPU.

**Comment by DqqHns** [comment]
> > > 我也有类似的问题。目前的实验表明，迭代次数取决于数据集的大小（你可以尝试使用较小的数据集来调整 max_steps）。论文中提到第一次训练需要 30 万次迭代。每个 epoch 的训练器使用 2331 张图像。我推断，在 MVTEC-AD 数据集上进行 30 万次迭代大约需要 128 个 epoch。
> > 
> > 
> > 好，不过我在单张3090上训练MVTEC-AD数据集，第一步（异常图像生成）就已经花了5天了，目前还在训练，大概到320epoch了，这有点太久了吧，你狮子为什么？
> 
> 您好，您可以通过设置 max_step 在 30w 次迭代时停止，这在单个 3090 GPU 上可能只需要大约 2-3 天。

您好，感谢您的回复，运行了5天，大约跑了320epoch，然后我自己手动退出然它停掉了，那这样可以直接进行后续操作嘛？

**Comment by DqqHns** [comment]
> > > 我也有类似的问题。目前的实验表明，迭代次数取决于数据集的大小（你可以尝试使用较小的数据集来调整 max_steps）。论文中提到第一次训练需要 30 万次迭代。每个 epoch 的训练器使用 2331 张图像。我推断，在 MVTEC-AD 数据集上进行 30 万次迭代大约需要 128 个 epoch。
> > 
> > 
> > 好，不过我在单张3090上训练MVTEC-AD数据集，第一步（异常图像生成）就已经花了5天了，目前还在训练，大概到320epoch了，这有点太久了吧，你狮子为什么？
> 
> 您好，您可以通过设置 max_step 在 30w 次迭代时停止，这在单个 3090 GPU 上可能只需要大约 2-3 天。

而且，我再问一下，后续训练其他的模块也是需要自己手动停止嘛？

**Comment by sjtuplayer** [comment]
> > > > 我也有类似的问题。目前的实验表明，迭代次数取决于数据集的大小（你可以尝试使用较小的数据集来调整 max_steps）。论文中提到第一次训练需要 30 万次迭代。每个 epoch 的训练器使用 2331 张图像。我推断，在 MVTEC-AD 数据集上进行 30 万次迭代大约需要 128 个 epoch。
> > > 
> > > 
> > > 好，不过我在单张3090上训练MVTEC-AD数据集，第一步（异常图像生成）就已经花了5天了，目前还在训练，大概到320epoch了，这有点太久了吧，你狮子为什么？
> > 
> > 
> > 您好，您可以通过设置 max_step 在 30w 次迭代时停止，这在单个 3090 GPU 上可能只需要大约 2-3 天。
> 
> 您好，感谢您的回复，运行了5天，大约跑了320epoch，然后我自己手动退出然它停掉了，那这样可以直接进行后续操作嘛？

Yes, you can end it manually. For the other training components, I remember that they can end automatically. You can have a try.

**Comment by DqqHns** [comment]
> > > > > 我也有类似的问题。目前的实验表明，迭代训练次数取决于数据集的大小（你可以尝试使用较小的数据集来调整 max_steps）。论文中提到第一次 30 万次迭代。每个 epoch 的训练器使用 2331 张图像。我推断，在 MVTEC-AD 数据集上进行 30 万次迭代大约需要 128 个 epoch。
> > > > 
> > > > 
> > > > 好，不过我在单张3090上训练MVTEC-AD数据集，第一步（异常图像生成）就已经花了5天了，目前还在训练，大概到320epoch了，这有点太久了吧，你狮子为什么？
> > > 
> > > 
> > > 您好，您可以通过设置 max_step 在 30w 次迭代时停止，这在单个 3090 GPU 上可能只需要大约 2-3 天。
> > 
> > 
> > 您好，感谢您的回复，运行了5天，大约跑了320epoch，然后我自己手动退出然后就停掉了，这样可以直接进行后续操作嘛？
> 
> 是的，你可以手动结束。其他训练部分我记得是可以自动结束的。你可以试试。

好的，非常感谢您的耐心解答
